include/pybind11/pybind11.h

```diff
@@ -1720,7 +1736,8 @@ template <return_value_policy Policy = return_value_policy::reference_internal,
           typename Sentinel,
           typename ValueType = decltype(*std::declval<Iterator>()),
           typename... Extra>
-iterator make_iterator(Iterator first, Sentinel last, Extra &&... extra) {
+detail::iterator_state<Iterator, Sentinel, false, Policy>
+[[deprecated]] make_iterator_ng(Iterator first, Sentinel last, Extra &&... extra) {
```

Comment:
I'm not 100% happy about deprecation. Probably the annotations are unnecessary and should be removed.